### ISSUE TITLE
Suggestion: Superset of the FSL

### ISSUE THREAD

**Issue by LanguageAgnostic** [getsentry/fsl.software#46]
Here’s a suggestion for a superset of the FSL. The idea is to have a license that could be ready to use for a wide range of businesses that care about open-source, free software **and** sustainability. 

**Goals:**
- Wider adoption
- Same sustainability values

## 1) Include an option for AGPL

This would satisfy a wider range of people on the open-source front by being both “free software” and “open-source”. Ideally, uses of this license would mean contributions back to the community, in a sustainability-wise fashion.

## 2) FSL 2.0: MIT-or-Apache

The "dual MIT / Apache-2.0 license" is quite popular in the Rust ecosystem, having many benefits: https://github.com/bevyengine/bevy/issues/2373. At this point, it could become the FSL-2.0 to make things more concise.

**Example:**
> All code in this repository is dual-licensed under either:
> - AGPL-3.0
> - FSL-2.0
>
> at your option.

The AGPL could also be directly offered (without the 2-year mark) in the FSL-2.0.

# Bonus

## 1) Explicit: Commercial licenses available

I believe there should be a commercial clause somewhere to make it possible for the company/creator to receive compensation in exchange for a valid commercial license (e.g., using the current code before the 2-year mark without any attribution).

*Inspired by Qt:*
> Licensees holding valid [PRODUCT] commercial licenses may use this software in accordance with the terms contained in a written agreement between you and [COMPANY]. Alternatively, the terms and conditions that were accepted by the licensee when buying and/or downloading the software do apply.

## 2) Explicit: Contributor License Agreement (CLA)

I believe there should also be an explicit CLA (could alternatively be recommended somewhere in documentation).

*Inspired by Bevy:*
> Unless you explicitly state otherwise, any contribution intentionally submitted for inclusion in the work by you, as defined in the Apache-2.0 license, shall be dual-licensed as above, without any additional terms or conditions.

## 3) Assets licenses

Some businesses (e.g., game development) might benefit from adding other licenses for assets, as they generally do not exactly fit with software code licenses. Ideally, these licenses should follow the spirit of the superset of the FSL. Suggestions are welcome!

# TLDR
- Free Software & Open-Source by FSF’s & OSI’s definitions (AGPL option)
- More use cases with MIT-or-Apache (a.k.a. FSL-2.0)
## BONUS
- Explicit: Commercial licenses available
- Explicit: Contributor License Agreement (CLA)
- Assets licenses (discussion)

Have a wonderful day!

**Comment by chadwhitacre** [comment]
Thanks for chiming in, @LanguageAgnostic. Do you have a concrete project in mind that you are actively looking to license like this or are you proposing this in the abstract?

**Comment by LanguageAgnostic** [comment]
Hi @chadwhitacre, I currently do have a few projects in prototyping, mainly in game development (funnily enough, an industry that could benefit from having more successful open-source products).

This is why I’m also looking to improve upon the FSL by including assets licensing, as it is common in creative industries. A suggestion that follows the spirit of the superset would be the following (open to discussion!):

- Provide the option of either **CC BY-SA** (*AGPL’s spirit*) or **CC BY-NC** (*FSL’s spirit*) before the 2-year mark and then **CC BY** (*MIT-or-Apache’s spirit*).

There could be a specific license that follows better the FSL's spirit of "*A Permitted Purpose is any purpose other than a Competing Use.*", but that has to be determined with the creative industries in mind.

Ideally, in the same fashion as the initial post, there should be a commercial clause to allow the company/creator to receive compensation in exchange for a valid commercial license (e.g., using the assets without any attribution).

Adding to the above content, one problem of the FSL license is that the software cannot be distributed in certain platforms that do promote open-source and free software (such as F-Droid, see inclusion policy: https://f-droid.org/en/docs/Inclusion_Policy/).

---

If you wish to use my company’s logo and upcoming website until the projects’ releases, I’ll be more than happy to provide. Otherwise, I’d rather view the superset of the FSL as applicable to a wider range of businesses than mine.

Thank you for reading the suggestions!

**Comment by chadwhitacre** [comment]
@LanguageAgnostic Are you aware of [the wider Fair Source initiative](https://fair.io/)? We're currently designing the website, which is just a stub right now. What would be wonderful from my perspective would be for you to build on the FSL for your particular use case, we can reference both from fair.io and we can build up a wider ecosystem of approaches under the bigger umbrella rather than trying to make FSL all things to all people. What do you think?

**Comment by LanguageAgnostic** [comment]
@chadwhitacre Seems like a good idea! Feel free to close this issue.

**Comment by chadwhitacre** [comment]
Awesome. Feel free to [open a ticket in the Fair Source repo](https://github.com/fairsource/fair.io/issues/new) when you've got something ready to go.
